subsys/net/ip/tcp.c

```diff
@@ -4734,6 +4752,39 @@ struct k_sem *net_tcp_conn_sem_get(struct net_context *context)
 	return &conn->connect_sem;
 }
 
+static void close_tcp_conn(struct tcp *conn, void *user_data)
+{
+	struct net_if *iface = user_data;
+	struct net_context *context = conn->context;
+
+	if (!net_context_is_used(context)) {
+		return;
+	}
+
+	if (net_context_get_iface(context) != iface) {
+		return;
+	}
+
+	k_mutex_lock(&context->lock, K_FOREVER);
+
+	if (IS_ENABLED(CONFIG_NET_OFFLOAD) && net_if_is_ip_offloaded(iface)) {
+		context->flags &= ~NET_CONTEXT_IN_USE;
+		(void)net_offload_put(iface, context);
+		k_mutex_unlock(&context->lock);
+		return;
```

Comment:
I noticed that this is return is not needed, we can have `else` and then unlock the lock in one place. I will update this.